src/platform/nrfconnect/OTAImageProcessorImpl.h

```diff
@@ -51,5 +50,45 @@ class OTAImageProcessorImpl : public OTAImageProcessorInterface
     uint8_t mBuffer[kBufferSize];
 };
 
+class ExtFlashHandler
+{
+public:
+    enum class Action : uint8_t
+    {
+        WAKE_UP,
+        SLEEP
+    };
+    virtual ~ExtFlashHandler() {}
+    virtual void DoAction(Action action);
+};
+
+class OTAImageProcessorImplPMDevice : public OTAImageProcessorImpl
+{
+public:
+    explicit OTAImageProcessorImplPMDevice(ExtFlashHandler & aHandler);
+    CHIP_ERROR PrepareDownload() override;
+    CHIP_ERROR Abort() override;
+    CHIP_ERROR Apply() override;
+
+private:
+    ExtFlashHandler & mHandler;
+};
+
+#ifdef CONFIG_CHIP_OTA_REQUESTOR
+namespace OTAImageProcessorNrf {
+// compile-time factory method
+inline OTAImageProcessorImpl & Get()
```

Comment:
I would move that to some (to-be-created) `OTAUtil.h` in `examples/platform/nrfconnect/`. The reason is that it's not generic enough. I can imagine customers using external flash for different purposes and then they wouldn't be able to use this function. Functions which have rather instrucitonal purpose and are likely not going to be used in production use-cases should not be part of the CHIP libary, imho.

Reply:
Agree, thanks for the hit.